msal-client-credentials-flow/README.md

```diff
@@ -0,0 +1,167 @@
+
+> :information_source: **Bold text** in the tables above matches (or is similar to) a UI element in the Azure portal, while `code formatting` indicates a value you enter into a text box in the Azure portal.
+
+### 2. Update code sample with app registration values
+
+In [_app.js_](app.js), update the Azure AD property values with those from your [app's registration in the Azure portal](https://docs.microsoft.com/azure/active-directory/develop/quickstart-configure-app-expose-web-apis).
+
+```javascript
+auth: {
+  // 'Application (client) ID' of app registration in Azure portal - this value is a GUID
+  clientId: '',
+
+  // Client secret 'Value' (not the ID) from 'Client secrets' in app registration in Azure portal
+  clientSecret: '',
+
+  // Full directory URL, in the form of https://login.microsoftonline.com/<tenant>
+  authority: ''
+  
+  // 'Object ID' of app registration in Azure portal - this value is a GUID
+  clientObjectId: ''
+}
+```
+
+### 3. Install package(s)
+
+Install the required Node.js and MSAL packages:
+
+```bash
+npm install
+```
+
+## Run the application
+
+```bash
+node app.js
+```
+
+If everything worked, you should receive a response from the downstream web API (Microsoft Graph, in this case) similar to this:
+
+```console
+$ curl http://localhost:8080/api/application
+{
+  "@odata.context": "https://graph.microsoft.com/v1.0/$metadata#applications/$entity",
+  "id": "003c4673-d4a0-4eb6-8979-d6269e08b87e",
+  "deletedDateTime": null,
+  "appId": "c085fc22-2484-41c0-b795-2e940f51df64",
+  "applicationTemplateId": null,
+  "disabledByMicrosoftStatus": null,
+  "createdDateTime": "2021-12-17T16:45:43Z",
+  "displayName": "Node Web API",
+  "description": null,
+  "groupMembershipClaims": null,
+  "identifierUris": ["api://c085fc22-2484-41c0-b795-2e940f51df64"],
+  "isDeviceOnlyAuthSupported": null,
+  "isFallbackPublicClient": true,
+  "notes": null,
+  "publisherDomain": "contoso.onmicrosoft.com",
+  "serviceManagementReference": null,
+  "signInAudience": "AzureADMyOrg",
+  "tags": [],
+  "tokenEncryptionKeyId": null,
+  "defaultRedirectUri": null,
+  "certification": null,
+  "optionalClaims": null
+  ...
+}
+```
+
+## About the code
+
+This Node.js web API has a single route (/api/application) that supports anonymous access.  When the anonymous route is called, the API requests its own application object from Microsoft Graph.
+
+This web API uses the Microsoft Authentication Library (MSAL) for Node.js to get an access token for Microsoft Graph as the application itself using the client credentials flow.  The `Express.js` framework is used to simplify building an API and routing of requests.  The `https` library for Node.js is used to issue the HTTP GET request to Microsoft Graph and to handle the response.
```

Comment:
```suggestion
This web API uses the following:

- [Microsoft Authentication Library (MSAL) for Node.js](https://github.com/AzureAD/microsoft-authentication-library-for-js/tree/dev/lib/msal-node) - Used by this web API to get an access token for Microsoft Graph. This API uses MSAL to execute the OAuth 2.0 client credentials grant flow to get the access token so it can use its own identity in its requests to the Microsoft Graph API.
- [Express](https://expressjs.com/) (Node.js web framework) - Minimizes the amount of code required to build the web API and route the requests made to it by client applications.
- [`https`](https://nodejs.org/api/https.html) (Node.js module) - For issuing "raw" HTTP `GET` requests to Microsoft Graph and handling the responses to those `GET` requests.
```

@skabou - Please adjust accordingly if I've misstated anything.

@Dickson-Mwendia @henrymbuguakiarie - This is more "tutorial article fodder" experimentation. We could consider standardizing on something like this, perhaps in a table format, for inclusion in both the tutorial articles and READMEs if we land on something clear and succinct enough.